tutorials/quickstart-python.md

```diff
@@ -0,0 +1,392 @@
+
+```python
+CONNECTION = "postgres://username:password@host:port/dbname?sslmode=require"
+```
+
+Alternatively you can specify each parameter in the connection string as follows
+
+
+```python
+CONNECTION = "dbname =tsdb user=tsdbadmin password=secret host=host.com port=5432 sslmode=require"
+```
+
+>:WARNING: The above method of composing a connection string is for test or development purposes only, for production purposes be sure to make sensitive details like your password, hostname, and port number environment variables.
+
+#### Step 3: Connect to Timescale database using the Psycopg2 connect function
+
+We’ll use the psycopg2 [connect function][psycopg2-connect] to create a new database session
+
+In your `main` function, add the following lines:
+
+```python
+def main():
+…
+	with psycopg2.connect(CONNECTION) as conn:
+		#Call the function that needs the database connection
+		func_1(conn)
+```
+
+Alternatively, you can create a connection object as follows and the pass that object 
+around as needed, like opening a cursor to perform database operations:
+
+```python
+def main():
+conn = psycopg2.connect(CONNECTION) 
+insert_data(conn)
+cur = conn.cursor()
+```
+
+Congratulations, you’ve successfully connected to TimescaleDB using Python!
+
+### Create a relational table [](create_table)
+
+#### Step 1: Formulate your SQL statement
+First, compose a string which contains the SQL state that you would use to create 
+a relational table. In the example below, we create a table called `sensors`, with 
+columns `id`, `type` and `location`:
+
+```python
+query_create_sensors_table = "CREATE TABLE sensors (id SERIAL PRIMARY KEY, type VARCHAR(50), location VARCHAR(50));"
+```
+
+#### Step 2: Execute the SQL statement and commit changes
+Next, we execute our `CREATE TABLE` statement by opening a cursor, executing the 
+query from Step 1 and committing the query we executed in order to make changes we 
+made to the database persistent. Afterward, we close the cursor we opened to clean 
+up:
+
+```python
+   cur = conn.cursor()
+   #see definition in Step 1
+   cur.execute(query_create_sensors_table)
+   conn.commit()
+   cur.close()
+```
+
+Congratulations, you’ve successfully created a relational table in TimescaleDB using Python!
+
+### Generate hypertable [](create_hypertable)
+
+In TimescaleDB, the primary point of interaction with your data is a [hypertable][hypertable], 
+the abstraction of a single continuous table across all space and time 
+intervals, such that one can query it via standard SQL.
+
+Virtually all user interactions with TimescaleDB are with hypertables. Creating tables 
+and indexes, altering tables, inserting data, selecting data, etc. can (and should) 
+all be executed on the hypertable.
+
+A hypertable is defined by a standard schema with column names and types, with at 
+least one column specifying a time value, and one (optional) column specifying an 
```

Comment:
You can have a hypertable that only consists of the time partitioning column, in general beginners should not start with additional partitioning columns as that puts additional constraints on the planner optimizations we can do.